### ISSUE TITLE
Initializers are not allowed in ambient contexts

### ISSUE THREAD

**Issue by rlasjunies** [alsatian-test/alsatian#251]
Hi @jamesrichford,

so far it's pretty going well, really like your work.

I've got one side effect that I do not like some much: the errors messages raised due to the constant declaration, in _metadata-keys.d.ts.

It could be great to avoid these errors. I don't think it's a compilation setup issue on my project. Do you have an idea?

![image](https://cloud.githubusercontent.com/assets/1439791/21539348/6fa93172-cda6-11e6-8376-df430067a47e.png)

**Comment by Jameskmonger** [comment]
@jamesrichford I thought that this would have been solved by the solution to #209.

@rlasjunies can you confirm the Alsatian version you've got installed?

**Comment by jamesrichford** [comment]
@jameskmonger - 209 is an issue with they source where as this is an issue with typings.

@rlasjunies - could we get a look at the repo or the `tsconfig.json` and TypeScript version you're using to recreate it?

**Comment by jamesrichford** [comment]
Thanks @rlasjunies I'll take a look in a couple of days when I'm back at a computer. Looks like your exclude should eliminate these sorts of errors though. 

How are you calling the TypeScript compiler out of interest? `tsc` via the CLI?

**Comment by rlasjunies** [comment]
Hi @jamesrichford 

The exclude was put in as a trial, to avoid these errors messages. Witth or without, it does nothing relevant. I've removed it.

I run `tsc -w` from msdos cmd. I've the same result via the command line of vscode.

The ts emit is made event with the error. 

Looking in the documentation, the message is quite explicit. "var initialisation should not be made in *.d.ts". But it's strange, that I've this error you don't. 

Again, it's not urgent.

**Comment by rlasjunies** [comment]
Hi Guys,

 This morning I do not have the error messages anymore. :-) My problem is that I do not know why it does not show :-(. 

Yesterday I've put "target": "es5" -> did not change. This evening I've close all my coding tools (vscode).

This morning, compilation run fine. put back again "target": "es2015" ... run fine!

Sorry to have bother ... If I have more info, I'll add to this thread.

You can close.

**Comment by jamesrichford** [comment]
Great news that it's working! Very strange though. I'm going to leave it open whilst I try to reproduce it though and let you know the results of my investigation

**Comment by jamesrichford** [comment]
So after a bit of digging I've found the cause of this. It seems that when TypeScript updated to 2.1 there was a breaking change in how declarations are output.

```
declare const TEST_FIXTURE: string
``` 

now gets output as 

```
declare const TEST_FIXTURE = "alsatian:test-fixture";
```

This causes an issue when a file has been compiled against TypeScript 2.1 (alsatian) is being used by a lower version of TypeScript (e.g. another library). So looks like you may have updated TypeScript which resulted in this issue disappearing for you.

Does this all add up for you?

**Comment by jamesrichford** [comment]
@rlasjunies :)

**Comment by rlasjunies** [comment]
@jamesrichford that a reasonnable explanation for my situation and changes made on my install :-)

Thank

**Comment by jamesrichford** [comment]
Excellent, I'll go ahead and close this then :)

**Comment by Jameskmonger** [comment]
Is our response if this crops up again just "update your TypeScript version" @jamesrichford?

**Comment by jamesrichford** [comment]
Will be considering a fix for backward compatibility, is this something you'd need @jameskmonger?

**Comment by jamesrichford** [comment]
@Jameskmonger @rlasjunies just raised issue #262 to fix this for 1.0.1
